kernel/arch/riscv/paging_generic.c

```diff
@@ -28,32 +28,150 @@ struct kmalloc_heap *hi_vmem_heap;
 
 void *ioremap(ulong paddr, size_t size)
 {
-   NOT_IMPLEMENTED();
+   ulong offset;
+   size_t count;
+   size_t page_count;
+   const u32 pg_flags = PAGING_FL_RW;
+
+   offset = paddr & (PAGE_SIZE - 1);
+   paddr -= offset;
+   size = pow2_round_up_at(size + offset, PAGE_SIZE);
+   page_count = size / PAGE_SIZE;
+
+   void *vaddr = hi_vmem_reserve(size);
+   if (!vaddr) {
+      printk("Unable to reserve hi vmem for ioremap() at %p\n", (void *)paddr);
+      return NULL;
+   }
+
+   count = map_kernel_pages(vaddr,
+                            paddr,
+                            page_count,
+                            pg_flags);
+
+   if (count < page_count) {
+
```

Comment:
Style: extra empty line (the following it much longer than the `if` statement).

Reply:
fixed！